lib/still/compiler/content_cache.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Still.Compiler.ContentCache do
```

Comment:
From what I understood this only caches the content from the file system. It can still cause multiple re-renders depending on the metadata.

I was wondering if we would notice significant improvements by doing the following:

1. Add a `compilation_pipeline` (or `compilation_stream` or `_chain`), etc to the `SourceFile` struct.
2. Add a hash of the metadata (similar to what is done in the `Profiler` to distinguish between different inclusions of the same file)
3. Save that as the key for rendering a file in a cache similar to this one.

Reply:
Yes, I was going to write about it in the description but I forgot. You're right, these changes only prevent recompilations, not re-renders. I didn't feel comfortable introducing that piece of code, it was too complex and I'm not sure if it was really going to speed things up. My idea was to wait until we have a big website available and run some benchmarks.

Reply:
That's a fair take.